Merge CAL menu

bwack opened the pull request:
If you like my changes with my CAL menu next to the CFG menu and the piecewise_map function used on adc_raw, please feel to merge :)

bwack committed:
Added CAL menu and piecewise interpolation

The CAL menu can change five (adc,temp) points which are used for
calibrating the temperature. Set ADC=1 in the CFG menu to read out adc
values. ADC values displayed are now calmed down with averaging.

bwack committed:
alphanumerics in display_string, x and y

bwack committed:
alphanumeric display_string

bwack commented on youyue858d.ino:
```diff
@@ -106,6 +108,18 @@ CPARAM temp_averages = { 100, 999, TEMP_AVERAGES_DEFAULT, TEMP_AVERAGES_DEFAULT,
 CPARAM slp_timeout = { 0, 30, SLP_TIMEOUT_DEFAULT, SLP_TIMEOUT_DEFAULT, 16, 17 };
 CPARAM fan_only = { 0, 1, 0, 0, 26, 27 };
 CPARAM display_adc_raw = { 0, 1, 0, 0, 28, 29 };
+CPARAM cal_x[] = {
```
maybe it should have been cal_adc[] and cal_temp[]

madworm replied:
You can still change it.

I won't get a chance to look at it before the weekend.

bwack commented on youyue858d.ino:
```diff
@@ -310,23 +347,57 @@ int main(void)
+					change_config_parameter(&cal_y[2], "Y2");
+					change_config_parameter(&cal_y[3], "Y3");
+					change_config_parameter(&cal_y[4], "Y4");
+        }
```
Same here, maybe it should ask for Ad0-Ad4 and T0-T4 for adc and temp values rather than x and y.

bwack committed:
Replaced x and y with adc and temp

bwack committed:
Cool the heater while wand is out of cradle after a restart

Cool the heater also when software ask for wand in cradle after a
restart. The heater might have been really hot right before the device
was power cycled or reset, and the fan should run if it asks for it to
be put back into the cradle.
